File: Partie-Mobile/lib/models/presence.dart
class presence {
  int? idPresnce;
  bool? absent;
  int? nombreAbsence;
  String? username;
  String? date;

  presence(
      {this.idPresnce,
      this.absent,
      this.nombreAbsence,
      this.username,
      this.date,});

  presence.fromJson(Map<String, dynamic> json) {
    idPresnce = json['idPresnce'];
    absent = json['absent'];
    nombreAbsence = json['nombreAbsence'];
    username = json['username'];
    date = json['date'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idPresnce'] = this.idPresnce;
    data['absent'] = this.absent;
    data['nombreAbsence'] = this.nombreAbsence;
    data['username'] = this.username;
    data['date'] = this.date;
    return data;
  }
}

File: Partie-Backend/src/main/java/com/basketapp/basket_app/models/Presence.java
package com.basketapp.basket_app.models;

import java.util.Date;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.PrePersist;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Presence {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private  Long idPresnce;
    private Boolean absent;
    private Integer nombreAbsence;
    private String username;
    @Temporal(TemporalType.TIMESTAMP)
     @Column(nullable = false)
     @JsonFormat(pattern="yyyy-MM-dd")
     private Date date;

     @PrePersist
     private void onCreate() {
        date = new Date();
    }

    
    
}
